app/src/main/java/com/blogspot/e_kanivets/moneytracker/ui/presenter/ShortSummaryPresenter.java

```diff
@@ -45,40 +49,67 @@ public ShortSummaryPresenter(Context context) {
         green = context.getResources().getColor(R.color.green);
     }
 
-    public View create(boolean shortSummary) {
+    public View create(boolean shortSummary, RecordAdapter.HeaderViewHolder mainViewHolder) {
         view = layoutInflater.inflate(R.layout.view_summary_records, null);
         view.findViewById(R.id.iv_more).setVisibility(shortSummary ? View.VISIBLE : View.INVISIBLE);
 
-        ViewHolder viewHolder = new ViewHolder(view);
-        view.setTag(viewHolder);
+        if (mainViewHolder == null)
+            view.setTag(new ViewHolder(view));
+        else
+            view.setTag(mainViewHolder);
 
         return view;
     }
 
     public void update(IRecordReport report, String currency, List<String> ratesNeeded) {
-        ViewHolder viewHolder = (ViewHolder) view.getTag();
-
-        if (report == null) {
-            viewHolder.tvTotalIncome.setText("");
-            viewHolder.tvTotalExpense.setText("");
-
-            viewHolder.tvTotal.setTextColor(red);
-            viewHolder.tvTotal.setText(createRatesNeededList(currency, ratesNeeded));
+        if (view.getTag() instanceof RecordAdapter.HeaderViewHolder) {
```

Comment:
Hm, you can extract an interface with `getTvTotalName(), ...` and then have both ViewHolders implementing them. In such a way you won't have duplication (and class check) here.